src/cogent3/core/new_alignment.py

```diff
@@ -286,9 +361,19 @@ def make_unaligned_seqs(
 
     if len(data) == 0:
         raise ValueError("data must be at least one sequence.")
-    seqs_data = prep_for_seqs_data(data, moltype)
 
-    return SequenceCollection(seqs_data=seqs_data, moltype=moltype, info=info)
+    seqs_data = make_seqsdata_dict(data, label_to_name=label_to_name)
+
+    if isinstance(seqs_data, dict):
```

Comment:
add a comment about design to revisit `make_unaligned_seqs()`. (Specifically I would do a single dispatch on `make_unaligned_seqs()` such that the version which takes a `SeqsData` instance does the checking on the alphabet consistency with the molecular type and then creates the sequence collection. All other types are all bundled together in one version that uses `make_seqsdata_dict()` and creates the `SeqsData` instance, and then calls `make_unaligned_seqs()` again.)

Reply:
Definitely agree to revisit, is evolving a bit randomly as functionality is added to SequenceCollection